code/game/machinery/bioprinter.dm

```diff
@@ -114,6 +114,13 @@
 	switch(action)
 		if("print")
 			var/recipe = params["recipe_id"]
+			var/valid_recipe = FALSE
+			for(var/datum/bioprinter_recipe/product_recipes in products)
+				if(product_recipes.type == text2path(recipe))
+					valid_recipe = TRUE
+					break
+			if(!valid_recipe)
```

Comment:
Maybe right out an admin message saying an illegal recipe was requested. Probably excessive but it would help monitor whats going on.